test/SparqlAntlrParserTestHelpers.h

```diff
@@ -392,9 +468,28 @@ MATCHER_P3(IsGraphPattern, optional, filters, childMatchers, "") {
          childrenMatch;
 }
 
-MATCHER_P2(IsSubSelect, selectMatcher, whereMatcher, "") {
-  auto query = std::get_if<GraphPatternOperation::Subquery>(&arg.variant_);
-  return query && query->_subquery.hasSelectClause() &&
-         testing::Value(query->_subquery.selectClause(), selectMatcher) &&
-         testing::Value(query->_subquery._rootGraphPattern, whereMatcher);
-}
+auto OptionalGraphPattern = [](vector<SparqlFilter>&& filters,
+                               const auto&... childMatchers) {
+  return IsOptional(
+      IsGraphPattern(true, filters, std::tuple{childMatchers...}));
+};
+
+auto MinusGraphPattern = [](vector<SparqlFilter>&& filters,
+                            const auto&... childMatchers) {
+  return IsMinus(IsGraphPattern(false, filters, std::tuple{childMatchers...}));
+};
+
+auto IsSubSelect = [](auto&& selectMatcher, auto&& whereMatcher) {
```

Comment:
My understanding of gtest is that Matchers normally are polymorphic with regards to the value that they can match. But it may also be worth a try whether it is possible to set the type a matcher can match.

Reply:
All except one Matcher lambda now has the type it matches explicitly specified. The type can now be directly read from the lambda's type. E.g. `Matcher<const GraphPatternOperation &> (*)(auto &&,auto &&) IsUnion`. This is done by explicitly specifying the return type in the lambda: `...-> testing::Matcher<const GraphPatternOperation&>...`.